tooling/profiler/src/cli/execution_flamegraph_cmd.rs

```diff
@@ -51,19 +50,21 @@ fn run_with_generator(
     let initial_witness = program.abi.encode(&inputs_map, None)?;
 
     println!("Executing");
-    let (_, profiling_samples) = nargo::ops::execute_program_with_profiling(
+    let (_, mut profiling_samples) = nargo::ops::execute_program_with_profiling(
         &program.bytecode,
         initial_witness,
         &Bn254BlackBoxSolver,
         &mut DefaultForeignCallExecutor::new(true, None, None, None),
     )?;
     println!("Executed");
 
-    let profiling_samples: Vec<Sample<FieldElement>> = profiling_samples
-        .into_iter()
+    println!("Collecting {} samples", profiling_samples.len());
+
+    let profiling_samples: Vec<BrilligExecutionSample> = profiling_samples
+        .iter_mut()
         .map(|sample| {
-            let call_stack = sample.call_stack;
-            let brillig_function_id = sample.brillig_function_id;
+            let call_stack = std::mem::take(&mut sample.call_stack);
+            let brillig_function_id = std::mem::take(&mut sample.brillig_function_id);
```

Comment:
I'm not clear on why this is preferable over just taking ownership over the samples. Do you know why?

Reply:
I had noticed we were hitting OOM errors during collection of our samples. It was difficult to confirm exactly but I believe `collect` is still allocating memory for `Vec<Sample>` and during collection our memory kept growing after execution. `take` also takes ownership, just it lets us to do it of part of our Vec, so I wanted to guarantee we were taking ownership of the sample rather than reallocating memory. This is not the same issue, but I had stumbled upon some articles that indicated issues with Rust's collect (https://blog.polybdenum.com/2024/01/17/identifying-the-collect-vec-memory-leak-footgun.html) as well. This led me to try indicating I want ownership of this sample data directly.